src/imgui/ImGuiReverseBar.cc

```diff
@@ -58,13 +62,23 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 		}
 		ImGui::Separator();
 
-		auto existingStates = manager.execute(TclObject("list_savestates"));
-		im::Menu("Load state ...", existingStates && !existingStates->empty(), [&]{
+		im::Menu("Load state ...", [&]{
 			im::Table("table", 2, ImGuiTableFlags_BordersInnerV, [&]{
 				if (ImGui::TableNextColumn()) {
 					ImGui::TextUnformatted("Select save state"sv);
 					im::ListBox("##list", ImVec2(ImGui::GetFontSize() * 20.0f, 240.0f), [&]{
-						for (const auto& name : *existingStates) {
+						std::vector<std::string> names;
+						for (auto context = userDataFileContext(STATE_DIR);
+						     const auto& path : context.getPaths()) {
+							foreach_file(path, [&](const std::string&, std::string_view name) {
+								if (name.ends_with(STATE_EXTENSION)) {
+									name.remove_suffix(STATE_EXTENSION.size());
+									names.emplace_back(std::string(name));
```

Comment:
Detail: replace this last line with `names.emplace_back(name);`   (caveat: not tested, but I think it'll work).

Background info:
In a `vector<T>` there are two main ways to insert elements at the back:
* `push_back(const T&)`: this makes a copy of the given element.
* `emplace_back(Args&&... args): this takes (a variable amount) of constructor parameters, and constructs the new element in-place in the back of the vector. This can save a copy.

So when you call `names.emplace_back(std::string(name))'`, you first create a (temporary) `string` object, and pass that to `emplace_back`. That temporary is used to construct another `string` (via the copy constructor).

If you instead call `names.emplace_back(name)`, you pass a `string_view` object to `emplace_back`. And that object is used to construct a `string`.

Reply:
I realize that, but I didn't try that, because the original code seemed to do the same thing: https://github.com/openMSX/openMSX/blob/bb18df8c864469563db24f95ea55a0baeee0db8f/src/imgui/ImGuiReverseBar.cc#L165 
Is this then a different situation?

Reply:
Yes, it's a different situation: this calls the constructor of `Names`. That one needs two `string` parameters. (And `string_view` does not _implicitly_ convert to `string`, only _explicitly_).

Anyway, this was only a detail. Initialization in c++ can be tricky (if you care about optimal code).